$MirOS: ports/www/php/patches/patch-ext_ftp_ftp_c,v 1.1.7.1 2005/03/18 16:00:00 tg Exp $
--- ext/ftp/ftp.c.orig	Thu Mar 17 17:16:27 2005
+++ ext/ftp/ftp.c	Wed May  4 14:38:06 2005
@@ -994,6 +994,7 @@ ftp_mdtm(ftpbuf_t *ftp, const char *path
 	struct tm	tm;
 	char		*ptr;
 	int		n;
+	int		y;
 
 	if (ftp == NULL) {
 		return -1;
@@ -1006,7 +1007,8 @@ ftp_mdtm(ftpbuf_t *ftp, const char *path
 	}
 	/* parse out the timestamp */
 	for (ptr = ftp->inbuf; *ptr && !isdigit(*ptr); ptr++);
-	n = sscanf(ptr, "%4u%2u%2u%2u%2u%2u", &tm.tm_year, &tm.tm_mon, &tm.tm_mday, &tm.tm_hour, &tm.tm_min, &tm.tm_sec);
+	n = sscanf(ptr, "%4u%2u%2u%2u%2u%2u", &y, &tm.tm_mon, &tm.tm_mday, &tm.tm_hour, &tm.tm_min, &tm.tm_sec);
+	tm.tm_year = y;
 	if (n != 6) {
 		return -1;
 	}
